helix-view/src/document.rs

```diff
@@ -1614,6 +1618,14 @@ impl Document {
         &self.selections
     }
 
+    pub fn view_data(&mut self, view_id: ViewId) -> &ViewData {
+        self.view_data.entry(view_id).or_default()
+    }
+
+    pub fn view_data_mut(&mut self, view_id: ViewId) -> &mut ViewData {
+        self.view_data.entry(view_id).or_default()
+    }
```

Comment:
I wonder if these should return `Option<&[mut ]Viewdata>` instead? That would allow us to drop the Default requirement for `ViewData`. It's not currently useful to do that between this and https://github.com/helix-editor/helix/pull/6198 though so I don't have a strong opinion either way